### ISSUE TITLE
WebRTCBin.createAnswer() callback not called sporadically (promise garbage collected)

### ISSUE THREAD

**Issue by jw-tecuri** [gstreamer-java/gst1-java-core#270]
Looks like the promise which is used inside the method sometimes gets garbage collected before callback is called.

reproducible with:
```
webRtcBin.createAnswer(answer -> {
   System.out.println("ANSWER");
});
System.gc();
```

**Comment by neilcsmith-net** [comment]
Yes, think I can see where the problem lies there. Most of the webrrc code was donated by someone who has not contributed for some time. Few options to consider to fix. Do you want to take it on?

**Comment by jw-tecuri** [comment]
See pull request

**Comment by neilcsmith-net** [comment]
> See pull request

Thanks.  That might be the simplest way to fix this, although I have a slight concern on that implementation.

As and when we know whether there will be any further releases of this library, I'll review and/or merge.

**Comment by jw-tecuri** [comment]
Extended the pull request with data channel support.
